atomisticparsers/gromacs/parser.py

```diff
@@ -690,6 +689,27 @@ def parse_thermodynamic_data(self):
                     sec_energy.contributions.append(
                         EnergyEntry(kind=self._metainfo_mapping[key], value=val))
 
+        if not flag_system_time_map:
+            if n_evaluations == len(sec_system):
+                for i_calc, calc in enumerate(sec_run.calculation):
+                    calc.system_ref = sec_system[i_calc]
+                self.logger.warning('Time and step information not available, but system and calculation are the same length.'
+                                    'Assuming correspondence and creating references between these two lists.')
+                return
+
+        for time, n in self._system_time_map.items():
+            sec_scc = sec_run.m_create(Calculation)
+            sec_scc.time = time
+            sec_scc.step = int((time / time_step).magnitude)
+
+        times = [calc.time for calc in sec_run.calculation]
```

Comment:
use calculation_times_ps, is it the same?

Reply:
no, it's not the same. times is now the full list of (non-overlapping) times from the thermo list (which is the same as calculation_times_ps) and the system list, if times are available. But perhaps there is still a shortcut that I am not seeing?